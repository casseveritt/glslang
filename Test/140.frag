#version 140

varying vec4 v;

in vec4 i;
out vec4 o;

in float gl_ClipDistance[5];

void main()
{
    float clip = gl_ClipDistance[2];
}
#ifdef GL_ES
#error GL_ES is set
#else
#error GL_ES is not set
#endif

in struct S { float f; } s; // ERROR

float patch = 3.1;