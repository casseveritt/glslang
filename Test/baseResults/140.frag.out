140.frag
Warning, version 140 is not yet complete; most features are present, but a few are missing.
WARNING: 0:3: varying deprecated in version 130; may be removed in future release
ERROR: 0:17: '#error' : GL_ES is not set  
ERROR: 0:20: 'fragment-shader struct input' : not supported for this version or the enabled extensions 
ERROR: 2 compilation errors.  No code generated.


ERROR: node is still EOpNull!
0:10  Function Definition: main( (void)
0:10    Function Parameters: 
0:12    Sequence
0:12      Sequence
0:12        move second child to first child (float)
0:12          'clip' (float)
0:12          direct index (smooth float)
0:12            'gl_ClipDistance' (smooth in 5-element array of float)
0:12            Constant:
0:12              2 (const int)
0:22  Sequence
0:22    move second child to first child (float)
0:22      'patch' (float)
0:22      Constant:
0:22        3.100000
0:?   Linker Objects
0:?     'v' (smooth in 4-component vector of float)
0:?     'i' (smooth in 4-component vector of float)
0:?     'o' (out 4-component vector of float)
0:?     'gl_ClipDistance' (smooth in 5-element array of float)
0:?     'gl_ClipDistance' (smooth in 5-element array of float)
0:?     's' (smooth in structure{float f})
0:?     'patch' (float)


Linked fragment stage:



